doc/content/source/userobjects/ClusteringUserObject.md

```diff
@@ -0,0 +1,22 @@
+# ClusteringUserObject
+
+## Description
+
+`ClusteringUserObject` works as the base class in Cardinal for indentifying clusters in mesh results, where it may be beneficial to apply
+mesh-tally amalgamation both in post processing or on the fly. ClusteringUserObject inherits from 
+[GeneralUserObject](GeneralUserObject.md). It implements the mesh walking process. This base class assumes that only active
+elements, sharing at least one common side can be clustered together based on heuristics that will be implemented 
+in user objects derived from this class. 
+
+!listing /ClusteringUserObject.h start=doco-user-object-interface-begin end=doco-user-object-interface-end include-start=false
```

Comment:
In documentation, we don't typically show the source code unless there's something unique we are trying to show. The documentation is moreso geared towards the user (who theoretically shouldn't need to know how the source code is structured). Instead, with documentation we prefer to show snippets of input files and describe how an object is being used in an input file. I don't think this PR has any input files yet (they can come later), so I'd suggest we remove this code block altogether and make a note that when the PR into Cardinal comes later, to have an `## Examples` markdown section.